pkg/client/clientset/versioned/typed/gadget/v1alpha1/trace.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by client-gen. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	"context"
+	"time"
+
+	v1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/apis/gadget/v1alpha1"
+	scheme "github.com/kinvolk/inspektor-gadget/pkg/client/clientset/versioned/scheme"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	watch "k8s.io/apimachinery/pkg/watch"
+	rest "k8s.io/client-go/rest"
+)
+
+// TracesGetter has a method to return a TraceInterface.
+// A group's client should implement this interface.
+type TracesGetter interface {
+	Traces(namespace string) TraceInterface
+}
+
+// TraceInterface has methods to work with Trace resources.
+type TraceInterface interface {
+	Create(ctx context.Context, trace *v1alpha1.Trace, opts v1.CreateOptions) (*v1alpha1.Trace, error)
+	Update(ctx context.Context, trace *v1alpha1.Trace, opts v1.UpdateOptions) (*v1alpha1.Trace, error)
+	UpdateStatus(ctx context.Context, trace *v1alpha1.Trace, opts v1.UpdateOptions) (*v1alpha1.Trace, error)
+	Delete(ctx context.Context, name string, opts v1.DeleteOptions) error
+	DeleteCollection(ctx context.Context, opts v1.DeleteOptions, listOpts v1.ListOptions) error
+	Get(ctx context.Context, name string, opts v1.GetOptions) (*v1alpha1.Trace, error)
+	List(ctx context.Context, opts v1.ListOptions) (*v1alpha1.TraceList, error)
+	Watch(ctx context.Context, opts v1.ListOptions) (watch.Interface, error)
+	Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts v1.PatchOptions, subresources ...string) (result *v1alpha1.Trace, err error)
+	TraceExpansion
+}
+
+// traces implements TraceInterface
+type traces struct {
```

Comment:
I am not sure if this code is generated, but why `traces` instead of `trace`?

Reply:
I don't know the specific answer but those APIs are named with the plural.

```
traceClient.GadgetV1alpha1().Traces("gadget")
clientset.CoreV1().Pods("default")
```

etc.

Reply:
If this code is generated, I am OK with this, as you do not have a lot of control on generated code.
More generally, can you please indicate (in a comment here) which parts of the code you write and which were generated?

Reply:
Sure, everything under pkg/client was generated. Btw, all generated files have a `Code generated by client-gen` comment after the license header.

Reply:
Oh! I missed this comments :sweat_smile:.
Thank you for the precision!